src/utils.rs

```diff
@@ -190,21 +197,85 @@ pub fn collect_prob_dist<E: Event>(
         }
 
         let variants = try!(utils::collect_variants(&mut record, false, false, None, false));
-        let tag = event.tag_name("PROB");
-        let event_probs = try!(record.info(tag.as_bytes()).float());
-        if let Some(event_probs) = event_probs {
-            // tag present
-            for (variant, event_prob) in variants.into_iter().zip(event_probs.into_iter()) {
-                if let Some(variant) = variant {
-                    if !variant.is_type(vartype) || event_prob.is_nan() {
-                        continue;
+        let mut events_prob_sum = LogProb::ln_zero();
+        for tag in &tags {
+            if let Some(event_probs) = try!(record.info(tag.as_bytes()).float()) {
```

Comment:
Instead of an if clause, please use a try-statement as previously, or better, the new ?-operator.

Reply:
Hmm, I'm using a `try!` macro here and the `Some()`, just like before -- I only combined them into a single statement. I could however replace the `try!` with the `?`-operator. Or am I missing something in your request?

Reply:
Oh sorry, I did not read carefully enough. Yes, it is ok like this. Maybe just replace the `try!` with the `?`-operator.